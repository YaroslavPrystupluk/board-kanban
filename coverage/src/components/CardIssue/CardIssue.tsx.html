
<!doctype html>
<html lang="en">

<head>
    <title>Code coverage report for src/components/CardIssue/CardIssue.tsx</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../prettify.css" />
    <link rel="stylesheet" href="../../../base.css" />
    <link rel="shortcut icon" type="image/x-icon" href="../../../favicon.png" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../sort-arrow-sprite.png);
        }
    </style>
</head>
    
<body>
<div class='wrapper'>
    <div class='pad1'>
        <h1><a href="../../../index.html">All files</a> / <a href="index.html">src/components/CardIssue</a> CardIssue.tsx</h1>
        <div class='clearfix'>
            
            <div class='fl pad1y space-right2'>
                <span class="strong">100% </span>
                <span class="quiet">Statements</span>
                <span class='fraction'>93/93</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">50% </span>
                <span class="quiet">Branches</span>
                <span class='fraction'>1/2</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">100% </span>
                <span class="quiet">Functions</span>
                <span class='fraction'>1/1</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">100% </span>
                <span class="quiet">Lines</span>
                <span class='fraction'>93/93</span>
            </div>
        
            
        </div>
        <p class="quiet">
            Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
        </p>
        <template id="filterTemplate">
            <div class="quiet">
                Filter:
                <input oninput="onInput()" type="search" id="fileSearch">
            </div>
        </template>
    </div>
    <div class='status-line high'></div>
    <pre><table class="coverage">
<tr><td class="line-count quiet"><a name='L1'></a><a href='#L1'>1</a>
<a name='L2'></a><a href='#L2'>2</a>
<a name='L3'></a><a href='#L3'>3</a>
<a name='L4'></a><a href='#L4'>4</a>
<a name='L5'></a><a href='#L5'>5</a>
<a name='L6'></a><a href='#L6'>6</a>
<a name='L7'></a><a href='#L7'>7</a>
<a name='L8'></a><a href='#L8'>8</a>
<a name='L9'></a><a href='#L9'>9</a>
<a name='L10'></a><a href='#L10'>10</a>
<a name='L11'></a><a href='#L11'>11</a>
<a name='L12'></a><a href='#L12'>12</a>
<a name='L13'></a><a href='#L13'>13</a>
<a name='L14'></a><a href='#L14'>14</a>
<a name='L15'></a><a href='#L15'>15</a>
<a name='L16'></a><a href='#L16'>16</a>
<a name='L17'></a><a href='#L17'>17</a>
<a name='L18'></a><a href='#L18'>18</a>
<a name='L19'></a><a href='#L19'>19</a>
<a name='L20'></a><a href='#L20'>20</a>
<a name='L21'></a><a href='#L21'>21</a>
<a name='L22'></a><a href='#L22'>22</a>
<a name='L23'></a><a href='#L23'>23</a>
<a name='L24'></a><a href='#L24'>24</a>
<a name='L25'></a><a href='#L25'>25</a>
<a name='L26'></a><a href='#L26'>26</a>
<a name='L27'></a><a href='#L27'>27</a>
<a name='L28'></a><a href='#L28'>28</a>
<a name='L29'></a><a href='#L29'>29</a>
<a name='L30'></a><a href='#L30'>30</a>
<a name='L31'></a><a href='#L31'>31</a>
<a name='L32'></a><a href='#L32'>32</a>
<a name='L33'></a><a href='#L33'>33</a>
<a name='L34'></a><a href='#L34'>34</a>
<a name='L35'></a><a href='#L35'>35</a>
<a name='L36'></a><a href='#L36'>36</a>
<a name='L37'></a><a href='#L37'>37</a>
<a name='L38'></a><a href='#L38'>38</a>
<a name='L39'></a><a href='#L39'>39</a>
<a name='L40'></a><a href='#L40'>40</a>
<a name='L41'></a><a href='#L41'>41</a>
<a name='L42'></a><a href='#L42'>42</a>
<a name='L43'></a><a href='#L43'>43</a>
<a name='L44'></a><a href='#L44'>44</a>
<a name='L45'></a><a href='#L45'>45</a>
<a name='L46'></a><a href='#L46'>46</a>
<a name='L47'></a><a href='#L47'>47</a>
<a name='L48'></a><a href='#L48'>48</a>
<a name='L49'></a><a href='#L49'>49</a>
<a name='L50'></a><a href='#L50'>50</a>
<a name='L51'></a><a href='#L51'>51</a>
<a name='L52'></a><a href='#L52'>52</a>
<a name='L53'></a><a href='#L53'>53</a>
<a name='L54'></a><a href='#L54'>54</a>
<a name='L55'></a><a href='#L55'>55</a>
<a name='L56'></a><a href='#L56'>56</a>
<a name='L57'></a><a href='#L57'>57</a>
<a name='L58'></a><a href='#L58'>58</a>
<a name='L59'></a><a href='#L59'>59</a>
<a name='L60'></a><a href='#L60'>60</a>
<a name='L61'></a><a href='#L61'>61</a>
<a name='L62'></a><a href='#L62'>62</a>
<a name='L63'></a><a href='#L63'>63</a>
<a name='L64'></a><a href='#L64'>64</a>
<a name='L65'></a><a href='#L65'>65</a>
<a name='L66'></a><a href='#L66'>66</a>
<a name='L67'></a><a href='#L67'>67</a>
<a name='L68'></a><a href='#L68'>68</a>
<a name='L69'></a><a href='#L69'>69</a>
<a name='L70'></a><a href='#L70'>70</a>
<a name='L71'></a><a href='#L71'>71</a>
<a name='L72'></a><a href='#L72'>72</a>
<a name='L73'></a><a href='#L73'>73</a>
<a name='L74'></a><a href='#L74'>74</a>
<a name='L75'></a><a href='#L75'>75</a>
<a name='L76'></a><a href='#L76'>76</a>
<a name='L77'></a><a href='#L77'>77</a>
<a name='L78'></a><a href='#L78'>78</a>
<a name='L79'></a><a href='#L79'>79</a>
<a name='L80'></a><a href='#L80'>80</a>
<a name='L81'></a><a href='#L81'>81</a>
<a name='L82'></a><a href='#L82'>82</a>
<a name='L83'></a><a href='#L83'>83</a>
<a name='L84'></a><a href='#L84'>84</a>
<a name='L85'></a><a href='#L85'>85</a>
<a name='L86'></a><a href='#L86'>86</a>
<a name='L87'></a><a href='#L87'>87</a>
<a name='L88'></a><a href='#L88'>88</a>
<a name='L89'></a><a href='#L89'>89</a>
<a name='L90'></a><a href='#L90'>90</a>
<a name='L91'></a><a href='#L91'>91</a>
<a name='L92'></a><a href='#L92'>92</a>
<a name='L93'></a><a href='#L93'>93</a>
<a name='L94'></a><a href='#L94'>94</a></td><td class="line-coverage quiet"><span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">6x</span>
<span class="cline-any cline-yes">6x</span>
<span class="cline-any cline-yes">6x</span>
<span class="cline-any cline-yes">6x</span>
<span class="cline-any cline-yes">6x</span>
<span class="cline-any cline-yes">6x</span>
<span class="cline-any cline-yes">6x</span>
<span class="cline-any cline-yes">6x</span>
<span class="cline-any cline-yes">6x</span>
<span class="cline-any cline-yes">6x</span>
<span class="cline-any cline-yes">6x</span>
<span class="cline-any cline-yes">6x</span>
<span class="cline-any cline-yes">6x</span>
<span class="cline-any cline-yes">6x</span>
<span class="cline-any cline-yes">6x</span>
<span class="cline-any cline-yes">6x</span>
<span class="cline-any cline-yes">6x</span>
<span class="cline-any cline-yes">6x</span>
<span class="cline-any cline-yes">6x</span>
<span class="cline-any cline-yes">6x</span>
<span class="cline-any cline-yes">6x</span>
<span class="cline-any cline-yes">6x</span>
<span class="cline-any cline-yes">6x</span>
<span class="cline-any cline-yes">6x</span>
<span class="cline-any cline-yes">6x</span>
<span class="cline-any cline-yes">6x</span>
<span class="cline-any cline-yes">6x</span>
<span class="cline-any cline-yes">6x</span>
<span class="cline-any cline-yes">6x</span>
<span class="cline-any cline-yes">6x</span>
<span class="cline-any cline-yes">6x</span>
<span class="cline-any cline-yes">6x</span>
<span class="cline-any cline-yes">6x</span>
<span class="cline-any cline-yes">6x</span>
<span class="cline-any cline-yes">6x</span>
<span class="cline-any cline-yes">6x</span>
<span class="cline-any cline-yes">6x</span>
<span class="cline-any cline-yes">6x</span>
<span class="cline-any cline-yes">6x</span>
<span class="cline-any cline-yes">6x</span>
<span class="cline-any cline-yes">6x</span>
<span class="cline-any cline-yes">6x</span>
<span class="cline-any cline-yes">6x</span>
<span class="cline-any cline-yes">6x</span>
<span class="cline-any cline-yes">6x</span>
<span class="cline-any cline-yes">6x</span>
<span class="cline-any cline-yes">6x</span>
<span class="cline-any cline-yes">6x</span>
<span class="cline-any cline-yes">6x</span>
<span class="cline-any cline-yes">6x</span>
<span class="cline-any cline-yes">6x</span>
<span class="cline-any cline-yes">6x</span>
<span class="cline-any cline-yes">6x</span>
<span class="cline-any cline-yes">6x</span>
<span class="cline-any cline-yes">6x</span>
<span class="cline-any cline-yes">6x</span>
<span class="cline-any cline-yes">6x</span>
<span class="cline-any cline-yes">6x</span>
<span class="cline-any cline-yes">6x</span>
<span class="cline-any cline-yes">6x</span>
<span class="cline-any cline-yes">6x</span>
<span class="cline-any cline-yes">6x</span>
<span class="cline-any cline-yes">6x</span>
<span class="cline-any cline-yes">6x</span>
<span class="cline-any cline-yes">6x</span>
<span class="cline-any cline-yes">6x</span>
<span class="cline-any cline-yes">6x</span>
<span class="cline-any cline-yes">6x</span>
<span class="cline-any cline-yes">6x</span>
<span class="cline-any cline-yes">6x</span>
<span class="cline-any cline-yes">6x</span>
<span class="cline-any cline-yes">6x</span>
<span class="cline-any cline-yes">6x</span>
<span class="cline-any cline-yes">6x</span>
<span class="cline-any cline-yes">6x</span>
<span class="cline-any cline-yes">6x</span>
<span class="cline-any cline-yes">6x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import { FC } from "react";
import { Card } from "react-bootstrap";
import { intlFormatDistance } from "date-fns";
import { Iissues } from "../../model/Iissues";
import { Draggable } from "react-beautiful-dnd";
&nbsp;
import { CardWrapp } from "./styledCardItem";
&nbsp;
interface CardIssueProps {
	issue: Iissues;
	index: number;
}
&nbsp;
const Cardissue: FC&lt;CardIssueProps&gt; = ({ issue, index }) =&gt; {
	// const today = new Date();
	// const updatedate = new Date(issue.updated_at);
	// const diffInMs = today.getTime() - updatedate.getTime();
	// const diffInDays = Math.floor(diffInMs / (1000 * 60 * 60 * 24));
&nbsp;
	// const date =
	// 	diffInDays &gt; 0 ? `opened ${diffInDays} days ago` : `opened ${updatedate.toLocaleDateString()}`;
&nbsp;
	// const getRelativeTimeString = (date: Date | number, lang = navigator.language) =&gt; {
	// 	// Отримуєм час мілісекундах якщо число в мілісекундах то берем число, якщо ми отримали обєкт дата то застосовуєм data.getTime
	// 	const timeMs = typeof date === "number" ? date : date.getTime();
&nbsp;
	// 	//Визначаєм різницю в секундах з часом який зараз
	// 	const deltaSeconds = Math.round((timeMs - Date.now()) / 1000);
&nbsp;
	// 	//Маив який має мінутуб часБ день, тиждень, місяць, рік
	// 	const cutoffs = [60, 3600, 86400, 86400 * 7, 86400 * 30, 86400 * 365, Infinity];
&nbsp;
	// 	// юніти які ми використовуєм в форматуванні (rtf.format(1, units ="second"))
	// 	const units: Intl.RelativeTimeFormatUnit[] = [
	// 		"second",
	// 		"minute",
	// 		"hour",
	// 		"day",
	// 		"week",
	// 		"month",
	// 		"year",
	// 	];
&nbsp;
	// 	//Визначаєм індех units який будем використовувати шукаєм індекси якщо менше 60 буде секунди більше буде хвилина і так далі
	// 	const unitIndex = cutoffs.findIndex((cutoff) =&gt; cutoff &gt; Math.abs(deltaSeconds));
&nbsp;
	// 	//Перевіряєм unitIndex на теперішнє значення якщо 0 то будем брати завжди брати 1 і ділити на 1 сек
	// 	const divisor = unitIndex ? cutoffs[unitIndex - 1] : 1;
&nbsp;
	// 	const rtf = new Intl.RelativeTimeFormat(lang, {
	// 		numeric: "auto",
	// 	});
&nbsp;
	// 	return rtf.format(Math.floor(deltaSeconds / divisor), units[unitIndex]);
	// };
	// const date = getRelativeTimeString(new Date(issue.updated_at), "en");
&nbsp;
	// За допомогою бібліотеки npm install date-fns --save
	const date = intlFormatDistance(new Date(issue.updated_at), Date.now(), {
		locale: "en",
		numeric: "always",
		// style: "long",
		// localeMatcher: "lookup",
	});
&nbsp;
	return (
		&lt;Draggable draggableId={`${issue.id.toString()}`} key={issue.id} index={index}&gt;
			{(provided, snapshot) =&gt; (
				&lt;CardWrapp
					{...provided.draggableProps}
					{...provided.dragHandleProps}
					ref={provided.innerRef}&gt;
					&lt;Card.Body
						style={{
							borderRadius: "10px",
							background: snapshot.isDragging<span class="branch-0 cbranch-no" title="branch not covered" > ? "#49bcf8" : "</span>#eaf4fc",
						}}&gt;
						&lt;Card.Title&gt;{issue.title}&lt;/Card.Title&gt;
						&lt;Card.Text&gt;
							#{issue.number} {date}
						&lt;/Card.Text&gt;
						&lt;Card.Text&gt;
							{issue.user.type} | Comments: {issue.comments}
						&lt;/Card.Text&gt;
						&lt;Card.Text&gt;State: {issue.state}&lt;/Card.Text&gt;
					&lt;/Card.Body&gt;
				&lt;/CardWrapp&gt;
			)}
		&lt;/Draggable&gt;
	);
};
&nbsp;
export default Cardissue;
&nbsp;</pre></td></tr></table></pre>

                <div class='push'></div><!-- for sticky footer -->
            </div><!-- /wrapper -->
            <div class='footer quiet pad2 space-top1 center small'>
                Code coverage generated by
                <a href="https://istanbul.js.org/" target="_blank" rel="noopener noreferrer">istanbul</a>
                at 2023-08-29T10:57:11.760Z
            </div>
        <script src="../../../prettify.js"></script>
        <script>
            window.onload = function () {
                prettyPrint();
            };
        </script>
        <script src="../../../sorter.js"></script>
        <script src="../../../block-navigation.js"></script>
    </body>
</html>
    