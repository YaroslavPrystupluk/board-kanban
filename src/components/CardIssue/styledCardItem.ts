import { Card } from "react-bootstrap";
import styled from "styled-components";

const CardWrapp = styled(Card)`
	border-radius: 10px;
	box-shadow: 5px 5px 5px 2px grey;
	margin: 0 10px 8px 10px;
	cursor: grab;
	background: ${(props) => (props.isDragging ? "#49bcf8" : "#eaf4fc")};
`;

export { CardWrapp };
