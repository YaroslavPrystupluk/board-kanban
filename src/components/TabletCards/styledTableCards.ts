import styled, { css } from "styled-components";

const ScrollStyles = css`
	::-webkit-scrollbar {
		display: none;
	}
`;

const Container = styled.div`
	background-color: #f4f5f7;
	border: 1px solid gray;
	border-radius: 2.5px;
	width: 300px;
	height: 475px;
	overflow-y: scroll;
	-ms-overflow-style: none;
	scrollbar-width: none;
	overflow-y: scroll;
	${ScrollStyles}
`;

const Title = styled.h3`
	position: sticky;
	top: 0;
	padding: 8px;
	background-color: #6cb8f7;
	text-align: center;
	z-index: 100;
`;
const IssuesList = styled.div`
	padding: 3px;
	transition: background-color 0.2s ease;
	background-color: #f4f5f7;
	flex-grow: 1;
`;
export { Container, Title, IssuesList };
