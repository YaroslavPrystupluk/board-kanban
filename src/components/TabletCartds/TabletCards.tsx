import { FC, DragEvent } from "react";
import { useAppSelector } from "../../hooks/hook";
import CardItem from "../Card/Card";
import { Iissues } from "../../model/Iissues";

interface IMyDragEvent extends DragEvent<HTMLDivElement> {
  target: HTMLDivElement & { style: CSSStyleDeclaration };
}

interface TabletCardsProps {
  handleDragOver: (e: IMyDragEvent) => void;
  handleDragStart: (issue: Iissues) => void;
  handleDragEnd: (e: IMyDragEvent) => void;
  handleDrop: (e: IMyDragEvent, issue: Iissues) => void;
  sortCard: (a: Iissues, b: Iissues) => number;
}

const TabletCards: FC<TabletCardsProps> = ({
  handleDragOver,
  handleDragStart,
  handleDragEnd,
  handleDrop,
  sortCard,
}) => {
  const { issues } = useAppSelector((state) => state.issues);

  return (
    <div>
      {issues.sort(sortCard).map((issue) => (
        <div
          key={issue.id}
          draggable="true"
          onDragOver={(e: IMyDragEvent) => handleDragOver(e)}
          onDragLeave={(e: IMyDragEvent) => handleDragEnd(e)}
          onDragStart={() => handleDragStart(issue)}
          onDragEnd={(e: IMyDragEvent) => handleDragEnd(e)}
          onDrop={(e: IMyDragEvent) => handleDrop(e, issue)}
        >
          <CardItem item={issue} />
        </div>
      ))}
    </div>
  );
};

export default TabletCards;
