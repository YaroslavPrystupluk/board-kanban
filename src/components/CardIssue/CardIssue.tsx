import { FC } from "react";
import { Card } from "react-bootstrap";
import { Iissues } from "../../model/Iissues";
import { Draggable } from "react-beautiful-dnd";

import { CardWrapp } from "./styledCardItem";

interface CardIssueProps {
	issue: Iissues;
	index: number;
}

const CardIssue: FC<CardIssueProps> = ({ issue, index }) => {
	const today = new Date();
	const updatedate = new Date(issue.updated_at);
	const diffInMs = today.getTime() - updatedate.getTime();
	const diffInDays = Math.floor(diffInMs / (1000 * 60 * 60 * 24));

	const date =
		diffInDays > 0 ? `opened ${diffInDays} days ago` : `opened ${updatedate.toLocaleDateString()}`;

	return (
		<Draggable draggableId={`${issue.id.toString()}`} key={issue.id} index={index}>
			{(provided, snapshot) => (
				<CardWrapp
					{...provided.draggableProps}
					{...provided.dragHandleProps}
					ref={provided.innerRef}>
					<Card.Body
						style={{
							borderRadius: "10px",
							background: snapshot.isDragging ? "#49bcf8" : "#eaf4fc",
						}}>
						<Card.Title>{issue.title}</Card.Title>
						<Card.Text data-testid="date">
							#{issue.number} {date}
						</Card.Text>
						<Card.Text data-testid="user">
							{issue.user.type} | Comments: {issue.comments}
						</Card.Text>
					</Card.Body>
				</CardWrapp>
			)}
		</Draggable>
	);
};

export default CardIssue;
