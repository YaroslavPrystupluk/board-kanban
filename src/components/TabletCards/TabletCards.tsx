import { FC } from "react";
import { Droppable } from "react-beautiful-dnd";
import CardItem from "../CardItem/CardItem";

import { Container, Title, IssuesList } from "./styledTableCards";
import { Iissues } from "../../model/Iissues";

interface TabletCardsProps {
	title: string;
	id: string;
	issues: Iissues[];
}

const TabletCards: FC<TabletCardsProps> = ({ title, issues, id }) => {
	return (
		<Container>
			<Title>{title}</Title>
			<Droppable droppableId={id}>
				{(provided) => (
					<IssuesList ref={provided.innerRef} {...provided.droppableProps}>
						{issues.map((issue, index) => (
							<CardItem key={issue.id} issue={issue} index={index} />
						))}
						{provided.placeholder}
					</IssuesList>
				)}
			</Droppable>
		</Container>
	);
};

export default TabletCards;
