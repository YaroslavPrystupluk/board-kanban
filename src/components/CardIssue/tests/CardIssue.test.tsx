import { describe, it, expect, vi } from "vitest";
import { DragDropContext, Droppable, Draggable } from "react-beautiful-dnd";
import { render, screen } from "@testing-library/react";
import CardIssue from "../CardIssue";
import TabletCards from "../../TabletCards/TabletCards";

const mockIssue = {
	id: 1,
	number: 2,
	title: "Sample Issue",
	updated_at: new Date().toISOString(),
	user: {
		type: "User",
	},
	comments: "3",
};

const mockIndex = 0;

const onDragEndMock = vi.fn();

describe("testing component 'CardItem'", () => {
	it("should match the snapshot and drag and drop works", () => {
		const { container } = render(
			<DragDropContext onDragEnd={onDragEndMock}>
				<Droppable droppableId="droppable">
					{(provided) => (
						<TabletCards
							title="Some Title"
							issues={[mockIssue]}
							id="1"
							{...provided.draggableProps}
							{...provided.dragHandleProps}
							ref={provided.innerRef}>
							<Draggable
								draggableId={`${mockIssue.id.toString()}`}
								key={mockIssue.id}
								index={mockIndex}>
								{(provided) => (
									<CardIssue
										issue={mockIssue}
										index={mockIndex}
										{...provided.draggableProps}
										{...provided.dragHandleProps}
										ref={provided.innerRef}
									/>
								)}
							</Draggable>
						</TabletCards>
					)}
				</Droppable>
			</DragDropContext>,
		);

		expect(container).toMatchSnapshot();

		const draggableElement = screen.getByTestId("IssuesList");
		expect(draggableElement).toBeInTheDocument();
	});

	it("renders the correct date based on the provided values", () => {
		render(
			<DragDropContext onDragEnd={onDragEndMock}>
				<Droppable droppableId="droppable">
					{(provided) => (
						<TabletCards
							title="Some Title"
							issues={[mockIssue]}
							id="1"
							{...provided.draggableProps}
							{...provided.dragHandleProps}
							ref={provided.innerRef}>
							<Draggable
								draggableId={`${mockIssue.id.toString()}`}
								key={mockIssue.id}
								index={mockIndex}>
								{(provided) => (
									<CardIssue
										issue={mockIssue}
										index={mockIndex}
										{...provided.draggableProps}
										{...provided.dragHandleProps}
										ref={provided.innerRef}
									/>
								)}
							</Draggable>
						</TabletCards>
					)}
				</Droppable>
			</DragDropContext>,
		);
		const dateElement = screen.getByTestId("date");
		expect(dateElement).toBeInTheDocument();
	});
});
