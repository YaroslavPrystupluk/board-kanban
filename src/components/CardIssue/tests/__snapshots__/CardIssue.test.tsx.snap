// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`testing component 'CardItem' > should match the snapshot and drag and drop works 1`] = `
<div>
  <div
    class="sc-gTRrQi knKbOs"
  >
    <h3
      class="sc-ewnqHT kuAEn"
    >
      Some Title
    </h3>
    <div
      class="sc-fFGjHI bkgfzW"
      data-rbd-droppable-context-id="0"
      data-rbd-droppable-id="1"
      data-testid="IssuesList"
    >
      <div
        aria-describedby="rbd-hidden-text-0-hidden-text-0"
        class="sc-bgqQcB gRmaKt card"
        data-rbd-drag-handle-context-id="0"
        data-rbd-drag-handle-draggable-id="1"
        data-rbd-draggable-context-id="0"
        data-rbd-draggable-id="1"
        draggable="false"
        role="button"
        tabindex="0"
      >
        <div
          class="card-body"
          style="border-radius: 10px; background: rgb(234, 244, 252);"
        >
          <div
            class="card-title h5"
          >
            Sample Issue
          </div>
          <p
            class="card-text"
            data-testid="date"
          >
            #
            2
             
            opened 31.05.2023
          </p>
          <p
            class="card-text"
            data-testid="user"
          >
            User
             | Comments: 
            3
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
